string::back used (not supported on pre-C++11)

Ran into the following issue when trying to build `protobuf` 3.5.0 where `string::back` was used. The [`string::back` method only exists in C++11]( http://en.cppreference.com/w/cpp/string/basic_string/back ). Found [`back` used on this line]( https://github.com/google/protobuf/blob/9c197b7ecba6019b7c81c673cf100b2fad4033ad/src/google/protobuf/testing/googletest.cc#L135 ). Snippet of this error taken from [this log]( https://ci.appveyor.com/project/conda-forge/protobuf-feedstock/build/1.0.106/job/wlim812l64mbq5j3 ) shown below.

```
C:\bld\protobuf_1513724287833\work\protobuf-3.5.0\src\google\protobuf\testing\googletest.cc(135) : error C2039: 'back' : is not a member of 'std::basic_string<_Elem,_Traits,_Ax>'
        with
        [
            _Elem=char,
            _Traits=std::char_traits<char>,
            _Ax=std::allocator<char>
        ]
NMAKE : fatal error U1077: 'C:\PROGRA~2\MICROS~1.0\VC\bin\cl.exe' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\nmake.exe"' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 9.0\VC\BIN\nmake.exe"' : return code '0x2'
Stop.
```

--- a/src/google/protobuf/testing/googletest.cc
+++ b/src/google/protobuf/testing/googletest.cc
@@ -132,7 +132,7 @@ string GetTemporaryDirectoryName() {
   // with "\\?\") but these will be degenerate in the sense that you cannot
   // chdir into such directories (or navigate into them with Windows Explorer)
   // nor can you open such files with some programs (e.g. Notepad).
-  if (result.back() == '.') {
+  if (result[result.size() - 1] == '.') {
     result[result.size() - 1] = '_';
   }
   // On Win32, tmpnam() returns a file prefixed with '\', but which is supposed

